### PR TITLE
Add basic layout mixins

### PR DESCRIPTION
**Pull request by lyzadanger**
Part of #2245 

This PR adds four layout-related mixins:

* `row` and `column` for establishing flex containers
* `horizontal-rhythm` and `vertical-rhythm` for spacing items within a container

The consolidates the way we're establishing those containers and spacings...

### CODE REVIEW

File: src/sidebar/components/annotation-share-control.js [hunk 1]
```diff
@@ -126,7 +126,11 @@ function AnnotationShareControl({
               className="annotation-share-control__links"
             />
           </div>
-          <SvgIcon name="pointer" className="annotation-share-panel__arrow" />
+          <SvgIcon
+            name="pointer"
+            inline={true}
```
Comment (lyzadanger) [inline src/sidebar/components/annotation-share-control.js:131]: This needed the `inline` variant for proper layout.

File: src/sidebar/components/annotation.js [hunk 2]
```diff
@@ -122,18 +122,19 @@ function Annotation({
         />
       )}
 
+      {isEditing && (
```
Comment (lyzadanger) [inline src/sidebar/components/annotation.js:125]: Adjusting rhythm helped me see that these controls don't belong in the footer anyway.

File: src/sidebar/components/selection-tabs.js [hunk 3]
```diff
@@ -153,29 +154,25 @@ function SelectionTabs({ isLoading, settings }) {
...
-              Create one by selecting some text and clicking the{' '}
-              <SvgIcon
-                name="annotate"
-                inline={true}
-                className="selection-tabs__icon"
-              />{' '}
-              button.
-            </div>
-          )}
+      {!isLoading && showNotesUnavailableMessage && (
```
Comment (lyzadanger) [inline src/sidebar/components/selection-tabs.js:157]: This no longer renders an empty `div` if there is no message to display.

File: src/sidebar/components/selection-tabs.js [hunk 4]
```diff
@@ -30,26 +29,28 @@ function Tab({
...
-      title={title}
-      aria-label={title}
-      aria-selected={isSelected.toString()}
-    >
-      {children}
-      {count > 0 && !isWaitingToAnchor && (
-        <span className="selection-tabs__count"> {count}</span>
-      )}
-    </button>
+    <div>
```
Comment (LMS007) [inline src/sidebar/components/selection-tabs.js:32]: What is the div needed for?

File: src/styles/mixins/layout.scss [hunk 5]
```diff
@@ -11,3 +11,69 @@
...
+    margin-left: $size;
+  }
+}
+
+/**
+ * Establish a vertical (margin) rhythm for this element's immediate
+ * children (i.e. put equal space between children). Currently set to 1em.
+ * @FIXME Find workaround for SvgIcon exception
+ */
+@mixin vertical-rhythm {
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/styles/mixins/layout.scss:75]: You'll see a theme expressed in this PR and others related to this work: just-in-time implementation. Yep, we'll probably eventually want to have this mixin take a parameter, but it hasn't been needed _yet_, so I'll do it when I see the need!

Iterate, iterate, iterate.

File: src/styles/mixins/layout.scss [hunk 6]
```diff
@@ -11,3 +11,69 @@
...
+  @include flex(row, $justify, $align);
+}
+
+/**
+ * Establish a horizontal (margin) rhythm for this element's immediate
+ * children (i.e. put equal space between children).
+ *
+ * @param $size [5px] - Size of horizontal margin between child elements
+ */
+@mixin horizontal-rhythm($size: 5px) {
```
[the remark this answers is not in the document]
Comment (lyzadanger) [reply src/styles/mixins/layout.scss:64]: Can you elaborate?

The way this is implemented, it puts space _between_ the immediate children of the element. The fact that it's implemented as `margin-left` is a little misleading in terms of suggesting it's a left margin on all children...

File: src/sidebar/components/selection-tabs.js (hunk 4 above)
Comment (lyzadanger) [reply src/sidebar/components/selection-tabs.js:32]: I believe it is intended as a "neutral" layout container for the contained `button`. IIRC, its parent has a `horizontal-rhythm` applied and there were layout funnies with having the `button` elements be direct descendants.

**Comment by lyzadanger** [comment]
> I still there there are more places that could use the general rhythm mixin [...]

Yes: Please keep in mind while reviewing these PRs that they are super duper iterative: I'm knocking things down one by one in a just-in-time manner...you're going to see places where you'll think "gosh, we could extend this" or "this should be applied in more places," and you're right!

What I'd suggest is, at the end of this onslaught of changes, we step back and evaluate at that point for consistency.

I'm trying to strike a balance between care and thoroughness but also encapsulation and ease-of-landing, if that makes sense.

**Pull request merged**